addons/missileguidance/functions/fnc_seekerType_ARH.sqf

```diff
@@ -0,0 +1,104 @@
+        _target = objNull;
+        _lastTargetPollTime = CBA_missionTime;
+        _seekerStateParams set [4, _lastTargetPollTime];
+        private _distanceToExpectedTarget = _seekerMaxRange min ((getPosASL _projectile) vectorDistance _expectedTargetPos);
+        
+        // Simulate how much the seeker can see at the ground
+        private _projDir = vectorDir _projectile;
+        private _projYaw = getDir _projectile;
+        private _rotatedYaw = (+(_projDir select 0) * sin _projYaw) + (+(_projDir select 1) * cos _projYaw);
+        if (_rotatedYaw isEqualTo 0) then { _rotatedYaw = 0.001 };
+        private _projPitch = atan ((_projDir select 2) / _rotatedYaw);
+        private _a1 = abs _projPitch;
+        private _a2 = 180 - ((_seekerAngle / 2) + _a1);
+        private _seekerBaseRadiusAtGround = ACTIVE_RADAR_MINIMUM_SCAN_AREA max (_distanceToExpectedTarget / sin(_a2) * sin(_seekerAngle / 2));
+        private _seekerBaseRadiusAdjusted = linearConversion [0, _seekerBaseRadiusAtGround, (CBA_missionTime - _lastTimeSeen) * vectorMagnitude _lastKnownVelocity, ACTIVE_RADAR_MINIMUM_SCAN_AREA, _seekerBaseRadiusAtGround, false];
+        // Look in front of seeker for any targets
+        private _nearestObjects = nearestObjects [_expectedTargetPos, ["AllVehicles"], _seekerBaseRadiusAdjusted, false];
+
+        _nearestObjects = _nearestObjects apply {
+            // I check both Line of Sight versions to make sure that a single bush doesnt make the target lock dissapear but at the same time ensure that this can see through smoke. Should work 80% of the time
+            if ([_projectile, getPosASL _x, _seekerAngle] call ace_missileguidance_fnc_checkSeekerAngle && { ([_projectile, _x, true] call ace_missileguidance_fnc_checkLOS) || { ([_projectile, _x, false] call ace_missileguidance_fnc_checkLOS) } }) then {
+                _x
+            } else {
+                objNull
+            };
+        };
+        _nearestObjects = _nearestObjects select { !isNull _x };
+        
+        // Select closest object to the expected position to be the current radar target
+        if ((count _nearestObjects) <= 0) exitWith {
+            _projectile setMissileTarget objNull;
+            _expectedTargetPos
+        };
+        private _closestDistance = _seekerBaseRadiusAtGround;
+        {
+            if ((_x distance2d _expectedTargetPos) < _closestDistance) then {
+                _closestDistance = _x distance2d _expectedTargetPos;
+                _target = _x;
+            };
+        } forEach _nearestObjects;
+    };
+
+    _projectile setMissileTarget _target;
+} else {
+    // External radar homing
+    // if the target is in the remote targets for the side, whoever the donor is will "datalink" the target for the hellfire.
+    private _remoteTargets = listRemoteTargets side _shooter;
+    if ((_remoteTargets findIf { (_target in _x) && (_x#1 > 0) }) < 0) then {
```

Comment:
I think if target is not in list it should go active immediately?

Reply:
Sadly due to the way `listRemoteTargets` works the vehicle needs to have datalink enabled for them to appear in this list

Reply:
shoot and then turn radar off
- With data link enabled: missile immediately goes active
- With data link disabled: `listRemoteTargets is always []` so nothing happens until time hits _timeWhenActive

Reply:
ops, this is fine I was thinking it worked differently